lib/devise/parameter_sanitizer.rb

```diff
@@ -135,7 +135,19 @@ def cast_to_hash(params)
     end
 
     def default_params
-      @params.fetch(@resource_name, {})
+      if hashable_resource_params?
+        @params.fetch(@resource_name)
+      else
+        empty_params
+      end
+    end
+
+    def hashable_resource_params?
+      @params.to_unsafe_h[@resource_name].is_a?(Hash)
```

Comment:
We don't need to call  `to_unsafe_h` we can just `@params[@resource_name].respond_to?(:permit)`

Reply:
Good idea, thanks.
Fix it.